File: expectanim/src/main/java/com/github/florent37/expectanim/core/rotation/RotationFlipExpectationValue.java
package com.github.florent37.expectanim.core.rotation;

import android.support.annotation.Nullable;
import android.view.View;

/**
 * Created by Christian Ringshofer on 17/02/2017.
 * <p>
 * A container for storing the rotation values for the flip animation
 */
public class RotationFlipExpectationValue extends RotationExpectation {

    private final float mRotationX, mRotationY;

    /**
     * a new flip expectation value which can be used to flip the view on the x- and y- axis
     * at the same time
     *
     * @param rotationX the x-rotation value around the x axis in degrees
     * @param rotationY the y-rotation value around the y axis in degrees
     */
    public RotationFlipExpectationValue(float rotationX, float rotationY) {
        mRotationX = rotationX;
        mRotationY = rotationY;
    }

    @Override
    public Float getCalculatedRotation(View viewToMove) {
        return null;
    }

    @Override
    @Nullable
    public Float getCalculatedRotationX(View viewToMove) {
        return mRotationX;
    }

    @Override
    @Nullable
    public Float getCalculatedRotationY(View viewToMove) {
        return mRotationY;
    }

}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/rotation/RotationFlipExpectationValue.kt
package com.github.florent37.kotlin.pleaseanimate.core.rotation

import android.view.View

/**
 * Created by Christian Ringshofer on 17/02/2017.
 *
 *
 * A container for storing the rotation values for the flip animation
 */
class RotationFlipExpectationValue(private val mRotationX: Float, private val mRotationY: Float) : RotationExpectation() {

    /**
     * a new flip expectation value which can be used to flip the view on the x- and y- axis
     * at the same time
     *
     * @param rotationX the x-rotation value around the x axis in degrees
     * @param rotationY the y-rotation value around the y axis in degrees
     */

    override fun getCalculatedRotation(viewToMove: View): Float? {
        return null
    }

    override fun getCalculatedRotationX(viewToMove: View): Float? {
        return mRotationX
    }

    override fun getCalculatedRotationY(viewToMove: View): Float? {
        return mRotationY
    }

}
